internal/services/arckubernetes/arc_kubernetes_cluster_extension_resource.go

```diff
@@ -0,0 +1,358 @@
+		},
+	}
+}
+
+func (r ArcKubernetesClusterExtensionResource) Update() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 30 * time.Minute,
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			client := metadata.Client.ArcKubernetes.ExtensionsClient
+
+			id, err := extensions.ParseExtensionID(metadata.ResourceData.Id())
+			if err != nil {
+				return err
+			}
+
+			var model ArcKubernetesClusterExtensionModel
+			if err := metadata.Decode(&model); err != nil {
+				return fmt.Errorf("decoding: %+v", err)
+			}
+
+			properties := &extensions.PatchExtension{
+				Properties: &extensions.PatchExtensionProperties{},
+			}
+
+			if metadata.ResourceData.HasChange("configuration_protected_settings") {
+				properties.Properties.ConfigurationProtectedSettings = &model.ConfigurationProtectedSettings
+			}
+
+			if metadata.ResourceData.HasChange("configuration_settings") {
+				properties.Properties.ConfigurationSettings = &model.ConfigurationSettings
+			}
+
+			if err := client.UpdateThenPoll(ctx, *id, *properties); err != nil {
+				return fmt.Errorf("updating %s: %+v", *id, err)
+			}
+
+			return nil
+		},
+	}
+}
+
+func (r ArcKubernetesClusterExtensionResource) Read() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 5 * time.Minute,
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			client := metadata.Client.ArcKubernetes.ExtensionsClient
+
+			id, err := extensions.ParseExtensionID(metadata.ResourceData.Id())
+			if err != nil {
+				return err
+			}
+
+			resp, err := client.Get(ctx, *id)
+			if err != nil {
+				if response.WasNotFound(resp.HttpResponse) {
+					return metadata.MarkAsGone(id)
+				}
+
+				return fmt.Errorf("retrieving %s: %+v", *id, err)
+			}
+
+			state := ArcKubernetesClusterExtensionModel{
+				Name:      id.ExtensionName,
+				ClusterID: arckubernetes.NewConnectedClusterID(metadata.Client.Account.SubscriptionId, id.ResourceGroupName, id.ClusterName).ID(),
+			}
+
+			if model := resp.Model; model != nil {
+				if err = metadata.ResourceData.Set("identity", identity.FlattenSystemAssigned(model.Identity)); err != nil {
+					return fmt.Errorf("setting `identity`: %+v", err)
+				}
+
+				if properties := model.Properties; properties != nil {
+					var originalModel ArcKubernetesClusterExtensionModel
+					if err := metadata.Decode(&originalModel); err != nil {
+						return fmt.Errorf("decoding: %+v", err)
+					}
+
+					state.ConfigurationProtectedSettings = originalModel.ConfigurationProtectedSettings
+
+					if properties.ConfigurationSettings != nil {
+						state.ConfigurationSettings = *properties.ConfigurationSettings
+					}
+
+					if properties.CurrentVersion != nil {
+						state.CurrentVersion = *properties.CurrentVersion
+					}
+
+					if properties.ExtensionType != nil {
+						state.ExtensionType = *properties.ExtensionType
+					}
+
+					if properties.ReleaseTrain != nil {
+						state.ReleaseTrain = *properties.ReleaseTrain
+					}
```

Comment:
We can use the pointer package to condense these
```suggestion

					state.ConfigurationSettings = pointer.From(properties.ConfigurationSettings)

					state.CurrentVersion = pointer.From(properties.CurrentVersion)

					state.ExtensionType = pointer.From(properties.ExtensionType)

					state.ReleaseTrain = pointer.From(properties.ReleaseTrain)
```